### PR TITLE
Migrations for the deprecated allowPrivilegedContainers field in Provisioner database

### PR DESCRIPTION
**Pull request by mvshao**
**Description**
Privileged Containers was deprecated in Kubernetes >=1.25. For this reason, unnecessary entries in the database and in the source code should be removed.

Changes proposed in this pull request:

- added the migrations to the database 
- cleanup the code and tests, update the Provisioner database schema
- delete dependencies to deprecated `kyma-release` database table

**Related issue(s)**
See the issue -> https://some.internal.repo/kyma/backlog/issues/3154

### CODE REVIEW

File: resources/kcp/charts/provisioner/migrations/202303061731_nullable_tiller_column.down.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+BEGIN;
+ALTER TABLE kyma_release ALTER COLUMN tiller_yaml SET NOT NULL;
```
[the remark this answers is not in the document]
Comment (VOID404) [reply resources/kcp/charts/provisioner/migrations/202303061731_nullable_tiller_column.down.sql:2]: We had a call about this, it was suggested to remove the migration and remove the tests that work on this table. Outside of the tests, the table is no longer used.

cc: @akgalwas @koala7659

**Review by VOID404 — CHANGES_REQUESTED** [review]

File: resources/kcp/charts/provisioner/migrations/202303142114_remove_tiller_column.down.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+BEGIN;
+ALTER TABLE kyma_release ADD COLUMN tiller_yaml text NOT NULL;
```
Comment (VOID404) [inline resources/kcp/charts/provisioner/migrations/202303142114_remove_tiller_column.down.sql:2]: This won't work if the table has any rows, will it?

File: components/provisioner/internal/api/resolver_integration_with_gardener_test.go [hunk 3]
```diff
@@ -207,7 +206,7 @@ func TestProvisioning_ProvisionRuntimeWithDatabase(t *testing.T) {
 		require.NoError(t, err)
 	}()
 
-	kymaConfig := fixKymaGraphQLConfigInput()
+	//kymaConfig := gqlschema.KymaConfigInput{}
```
Comment (VOID404) [inline components/provisioner/internal/api/resolver_integration_with_gardener_test.go:209]: Remove, don't comment out
We have `git` if we need it back

File: components/provisioner/internal/api/resolver_integration_with_gardener_test.go [hunk 4]
```diff
@@ -259,14 +255,11 @@ func TestProvisioning_ProvisionRuntimeWithDatabase(t *testing.T) {
...
-			err = insertDummyReleaseIfNotExist(releaseRepository, uuidGenerator.New(), kymaVersion)
-			require.NoError(t, err)
-
-			fullConfig := gqlschema.ProvisionRuntimeInput{RuntimeInput: &runtimeInput, ClusterConfig: &clusterConfig, KymaConfig: kymaConfig}
+			fullConfig := gqlschema.ProvisionRuntimeInput{RuntimeInput: &runtimeInput, ClusterConfig: &clusterConfig}
 
 			testProvisionRuntime(t, ctx, resolver, fullConfig, config.runtimeID, shootInterface, secretsInterface, config.auditLogTenant)
 
-			testUpgradeRuntimeAndRollback(t, ctx, resolver, dbsFactory, config.runtimeID)
+			//testUpgradeRuntimeAndRollback(t, ctx, resolver, dbsFactory, config.runtimeID)
```
Comment (VOID404) [inline components/provisioner/internal/api/resolver_integration_with_gardener_test.go:262]: Remove, don't comment out
We have `git` if we need it back

File: components/provisioner/internal/api/resolver_integration_with_gardener_test.go [hunk 5]
```diff
@@ -364,13 +357,13 @@ func testProvisionRuntime(t *testing.T, ctx context.Context, resolver *api.Resol
 	}
 
 	assert.Equal(t, expectedSeed, *runtimeStatusProvisioned.RuntimeConfiguration.ClusterConfig.Seed)
-	assert.Equal(t, fixKymaGraphQLConfig(), runtimeStatusProvisioned.RuntimeConfiguration.KymaConfig)
+	//assert.Equal(t, &gqlschema.KymaConfig{} runtimeStatusProvisioned.RuntimeConfiguration.KymaConfig)
```
Comment (VOID404) [inline components/provisioner/internal/api/resolver_integration_with_gardener_test.go:360]: Remove, don't comment out

File: components/provisioner/internal/api/resolver_integration_with_gardener_test.go [hunk 6]
```diff
@@ -559,9 +552,10 @@ func testProvisioningTimeouts() queue.ProvisioningTimeouts {
 
 func testProvisioningNoInstallTimeouts() queue.ProvisioningNoInstallTimeouts {
 	return queue.ProvisioningNoInstallTimeouts{
-		ClusterCreation:  5 * time.Minute,
-		ClusterDomains:   5 * time.Minute,
-		BindingsCreation: 5 * time.Minute,
+		ClusterCreation:    5 * time.Minute,
+		ClusterDomains:     5 * time.Minute,
+		BindingsCreation:   5 * time.Minute,
+		AgentConfiguration: 5 * time.Minute,
```
Comment (VOID404) [inline components/provisioner/internal/api/resolver_integration_with_gardener_test.go:558]: magic value

Maybe the time could be extracted to a const?
Not important - I'll approve despite this

**Review by VOID404 — CHANGES_REQUESTED** [review]
Could you maybe fix all the `-name` -> `--name`?

File: components/provisioner/internal/api/resolver_integration_with_gardener_test.go (hunk 4 above)
Comment (mvshao) [reply components/provisioner/internal/api/resolver_integration_with_gardener_test.go:262]: Ok, I will delete this.

File: components/provisioner/internal/api/resolver_integration_with_gardener_test.go (hunk 5 above)
Comment (mvshao) [reply components/provisioner/internal/api/resolver_integration_with_gardener_test.go:360]: Ok, I will delete this.

**Review by VOID404 — APPROVED** [review]

**Review by mmitoraj — APPROVED** [review]

**Pull request merged**